File: Vanilla/sources/zdoc/media/lua/shared/Library/Language.lua
---@class Language : zombie.core.Language
---@field private index int
---@field private name String
---@field private text String
---@field private charset String
---@field private base String
---@field private azerty boolean
Language = {}

---@public
---@return int
function Language:index() end

---@public
---@return String
function Language:toString() end

---@public
---@return String
function Language:text() end

---@public
---@return boolean
function Language:isAzerty() end

---@public
---@return String
function Language:charset() end

---@public
---@param value int
---@return Language
function Language:fromIndex(value) end

---@public
---@return String
function Language:base() end

---@public
---@param str String
---@return Language
function Language:FromString(str) end

---@public
---@return String
function Language:name() end

File: Vanilla/sources/zomboid/zombie/core/Language.java
package zombie.core;

public final class Language {
   private final int index;
   private final String name;
   private final String text;
   private final String charset;
   private final String base;
   private final boolean azerty;

   Language(int var1, String var2, String var3, String var4, String var5, boolean var6) {
      this.index = var1;
      this.name = var2;
      this.text = var3;
      this.charset = var4;
      this.base = var5;
      this.azerty = var6;
   }

   public int index() {
      return this.index;
   }

   public String name() {
      return this.name;
   }

   public String text() {
      return this.text;
   }

   public String charset() {
      return this.charset;
   }

   public String base() {
      return this.base;
   }

   public boolean isAzerty() {
      return this.azerty;
   }

   public String toString() {
      return this.name;
   }

   public static Language fromIndex(int var0) {
      return Languages.instance.getByIndex(var0);
   }

   public static Language FromString(String var0) {
      Language var1 = Languages.instance.getByName(var0);
      if (var1 == null) {
         var1 = Languages.instance.getDefaultLanguage();
      }

      return var1;
   }
}
